src/index.ts

```diff
@@ -209,11 +215,22 @@ export function setTool(state: Tool) {
  * Calls the function to save the file.
  */
 async function saveMode() {
+    let name: string;
+    let data: AEGTree | AEGTree[];
+
+    if (treeContext.modeState === "Draw") {
+        name = "AEG Tree";
+        data = treeContext.tree;
+    } else {
+        name = "Proof History";
```

Comment:
Its not the proof history, it is the proof. We should name the files to correspond to the proofs they represent. for example, A proof starting with the graph `A` and ending with `(A(A))` due to a DC intro 2 iterations and an erasure would be named: `A=>(A(A)).json`